app/generate.js

```diff
@@ -131,19 +168,22 @@ async function generatePages(renderedComponents) {
     }
 
     if (!page.header) {
-      [page.header] = defaultHeaders;
+      page.header = defaultHeaders.find(header => header.lang.id === page.lang.id);
     } else {
       page.header.html = generateHeader({
         theme,
         orgName: name,
         logo: page.header.logo,
         logoDescription: page.header.logoDescription,
         headerTagline: page.header.tagline,
+        langMenu,
+        lang: page.header.lang,
+        defaultLang,
```

Comment:
Maybe spread `page.header` at the top instead of having to define each unique header property that you want to use?
Like this: ```
generateHeader({   
  ...page.header,  
  theme,  
  orgName: name,  
  langMenu,  
  defaultLang,  
});```